frame/assets/src/lib.rs

```diff
@@ -193,6 +201,12 @@ pub mod pallet {
 			+ MaxEncodedLen
 			+ TypeInfo;
 
+		/// Max number of items to destroy per extrinsic call.
+		/// This number should be less than the value that can fit in a block for the various
+		/// extrinsics
```

Comment:
```suggestion
		/// Max number of items to destroy per `destroy_accounts` and `destroy_approvals` call.
		///
		/// Must be configured to result in a weight that makes each call fit in a block.
```

Reply:
I think we need another limit directly in the calls: `destroy_accouts(max_items: u32)` and analogous for approvals.  
Otherwise this limit needs to close of a monitoring on weight updates to be close to the max block weight.  
And it would always result in huge extrinsics, which is not necessarily what you want since they would fill up the blocks too much.